salt/config/__init__.py

```diff
@@ -1343,6 +1346,7 @@ def _gather_buffer_space():
     'archive_jobs': False,
     'root_dir': salt.syspaths.ROOT_DIR,
     'pki_dir': os.path.join(salt.syspaths.CONFIG_DIR, 'pki', 'master'),
+    'key_dir': os.path.join(salt.syspaths.CONFIG_DIR, 'key'),
```

Comment:
This looks like it is going to change the default from where it is currently, yes? As I understand this issue, I believe that's the intention but just want to clarify.

Reply:
Yes, that is right. The main purpose of this PR is changing the base of key files. Currently, the key file is generated on 'cachedir', and it is hard-corded. But the key file shouldn't be shared with other masters. If the key file is overwritten by the other master by mounting same physical storage for 'cachedir' as a setting of multi-master, several problems happen (#42377, #42532).
Therefore through this commit, salt master can avoid the problems. (by adding an option about 'key_dir', and a default value of the option).

F.Y.I.
By my investigation, while starting up, salt master generates and caches the key files on memory, and uses the cached key file to compare payload's (from salt cli or others) key field. If there aren't any other usage of the key files, there is no reason to generate key files on 'cachedir'.

If you have any questions or opinions about the commit, please leave comments for me. I will gladly answer to that. 🙂

Reply:
Thanks for the clarification. I'm definitely concerned about doing this in a point release. How do you feel about moving this to the develop branch?

Reply:
Your opinion sounds reasonable to me too. I will move to develop branch. Thanks for your comment.